### ISSUE TITLE
Flatpak support?

### ISSUE THREAD

**Issue by shirshak55** [httptoolkit/httptoolkit#299]
I tried to find httptoolkit in flatpak https://flathub.org/apps/search/httptoolkit but it gave no result. So, I am assuming httptoolkit doesn't have flatpak support yet.

It would be nice if there is flatpak support, so we don't have to worry about updating applications etc. I did see some flatpak-related code in package.json, but I am unsure about it.

**Comment by pimterry** [comment]
This will be due to flatpak sandboxing, because to detect browsers (and most other apps) HTTP Toolkit needs to be able to read & execute files from your whole file system. On Linux, browsers are found by checking for certain binaries in your $PATH, and I expect this affects other things too like ABD & terminal interception.

I think it's possible to use flatpak but disable the FS sandbox, which should work. Want to give that a go? If we can make flatpak work as well as other formats then I'd be happy to include it as an install option.

**Comment by shirshak55** [comment]
@pimterry which $PATH? I have added chrome to .zshrc. So, how does it even works? When we launch httptoolkit from Activity Launcher generally these paths are not inherited?

**Comment by pimterry** [comment]
I'm not sure if $PATH is inherited inside Flatpak. I would guess not. I think it might not matter though - even if you add paths to your $PATH variable, the files themselves probably aren't readable or runnable from inside Flatpak.

The way Flatpak works is that it runs the application in a separate isolated filesystem, where the application in flatpak can't normally see any other files on your computer, or touch anything else in the system, unless it's been specifically allowed to do so. I haven't played with it much so I don't fully understand the internals, but that's the gist. https://docs.flatpak.org/en/latest/sandbox-permissions.html has more details.

**Comment by shirshak55** [comment]
@pimterry Yes, I think we should go with flatpak without fs sandbox. Also, even if the user denies fs permission at least we can use HTTP toolkit with manual configuration ( Setting proxy to HTTP toolkit IP and installing certs manually)?
